sycl/include/CL/sycl/detail/array.hpp

```diff
@@ -19,8 +19,9 @@ template <int dimensions> class range;
 namespace detail {
 
 template <int dimensions = 1> class array {
+  static_assert(dimensions >= 1, "Array cannot be 0-dimensional.");
 public:
-  array() : common_array{0} {}
+  array() : common_array{} {}
```

Comment:
I find current version more readable.

Reply:
Initially I had an error complaining on having too many initializers for arra<0>. Fixing it this way did not help as the error went to the linkage phase where SPIRV failed due to having an array with size=0. After I added static_assert to array class, I can revert this code to 'common_array{0}'.